### PR TITLE
feat: Integrate Aries framework (DIDComm) into adapter

### LINKED ISSUE #40: Integrate aries agent
Integrate aries agent - https://github.com/trustbloc/edge-adapter/issues/26

### PR DESCRIPTION
**Pull request by rolsonquadras**
closes #40 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: cmd/adapter-rest/startcmd/start.go [hunk 1]
```diff
@@ -84,6 +92,26 @@ const (
...
+	// inbound host external url flag
+	didCommInboundHostExternalFlagName  = "didcomm-inbound-host-external"
+	didCommInboundHostExternalEnvKey    = "ADAPTER_REST_DIDCOMM_INBOUND_HOST_EXTERNAL"
+	didCommInboundHostExternalFlagUsage = "Inbound Host External Name:Port." +
+		" This is the URL for the inbound server as seen externally." +
+		" If not provided, then the internal inbound host will be used here." +
+		" Alternatively, this can be set with the following environment variable: " + didCommInboundHostExternalEnvKey
+
+	// db path
+	didCommDBPathFlagName  = "didcomm-db-path"
```
Comment (fqutishat) [inline cmd/adapter-rest/startcmd/start.go:111]: not needed we will use sql storage in the future

File: test/bdd/features/healthcheck_api.feature [hunk 2]
```diff
@@ -11,8 +10,9 @@ Feature: health check
...
-    Then the JSON path "status" of the response equals "success"
+    Then the JSON path "<respKey>" of the response equals "<respKeyVal>"
     Examples:
-      | url                                     |
-      | http://localhost:8070/healthcheck       |
-      | http://localhost:8069/healthcheck       |
+      | url                                             | respKey       | respKeyVal                                      |
+      | http://localhost:8070/healthcheck               | status        | success                                         |
+      | http://localhost:8060/healthcheck               | status        | success                                         |
+      | http://localhost:8060/didcomm/invitation        | @type         | https://didcomm.org/didexchange/1.0/invitation  |
```
Comment (troyronda) [inline test/bdd/features/healthcheck_api.feature:18]: Will this become out-of-band?

File: cmd/adapter-rest/startcmd/start.go (hunk 1 above)
Comment (fqutishat) [reply cmd/adapter-rest/startcmd/start.go:111]: this flag needed just for leveldb

Comment (rolsonquadras) [reply cmd/adapter-rest/startcmd/start.go:111]: SQL store is not available at the moment. Created an issue to switch - https://github.com/trustbloc/edge-adapter/issues/42.

File: test/bdd/features/healthcheck_api.feature (hunk 2 above)
Comment (rolsonquadras) [reply test/bdd/features/healthcheck_api.feature:18]: Yeah, internally gonna switch from `didexchange` to `oob`. For now, added generic `didcomm` term in the path.

File: pkg/restapi/issuer/operation/operations.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ import (
 
 const (
 	// API endpoints
-	generateInvitationEndpoint = "/didexchange/invitation"
+	generateInvitationEndpoint = "/didcomm/invitation"
```
Comment (llorllale) [inline pkg/restapi/issuer/operation/operations.go:23]: Who calls this endpoint?

I was thinking of going a slightly different route on the RP side: we currently have an endpoint that returns the presentation-definition. I think we should bundle the didcomm invitation in there as well, so the frontend makes only one call.

Comment (rolsonquadras) [reply pkg/restapi/issuer/operation/operations.go:23]: This will be called by issuer.

**Pull request merged**